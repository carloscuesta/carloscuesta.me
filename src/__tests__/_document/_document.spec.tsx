import renderer from 'react-test-renderer'

import Document from 'src/pages/_document'

jest.mock('next/document', () => ({
  __esModule: true,
  default: require('react').Component,
  Head: ({ children }) => <head>{children}</head>,
  Html: ({ children }) => children,
  Main: () => <main />,
  NextScript: () => 'NextScript'
}))

describe('_document', () => {
  it('should match _document', () => {
    const wrapper = renderer.create(<Document />)

    expect(wrapper).toMatchSnapshot()
  })
})
