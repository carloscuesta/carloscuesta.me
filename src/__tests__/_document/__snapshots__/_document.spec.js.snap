// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`_document should match _document 1`] = `
[
  <head>
    <meta
      content="78vmlhi_erc-UGybxcGwHyiUtf04wzYExTLa-4LoWio"
      name="google-site-verification"
    />
    <meta
      content="/ms-icon-144x144.png"
      name="msapplication-TileImage"
    />
    <link
      href="https://www.google-analytics.com"
      rel="preconnect"
    />
    <link
      href="https://www.google-analytics.com"
      rel="dns-prefetch"
    />
    <link
      href="/favicon/apple-icon-114x114.png"
      rel="apple-touch-icon"
      sizes="114x114"
    />
    <link
      href="/favicon/apple-icon-120x120.png"
      rel="apple-touch-icon"
      sizes="120x120"
    />
    <link
      href="/favicon/apple-icon-144x144.png"
      rel="apple-touch-icon"
      sizes="144x144"
    />
    <link
      href="/favicon/apple-icon-152x152.png"
      rel="apple-touch-icon"
      sizes="152x152"
    />
    <link
      href="/favicon/apple-icon-180x180.png"
      rel="apple-touch-icon"
      sizes="180x180"
    />
    <link
      href="/favicon/apple-icon-57x57.png"
      rel="apple-touch-icon"
      sizes="57x57"
    />
    <link
      href="/favicon/apple-icon-60x60.png"
      rel="apple-touch-icon"
      sizes="60x60"
    />
    <link
      href="/favicon/apple-icon-72x72.png"
      rel="apple-touch-icon"
      sizes="72x72"
    />
    <link
      href="/favicon/apple-icon-76x76.png"
      rel="apple-touch-icon"
      sizes="76x76"
    />
    <link
      href="/favicon/favicon-16x16.png"
      rel="icon"
      sizes="16x16"
      type="image/png"
    />
    <link
      href="/favicon/icon-192x192.png"
      rel="icon"
      sizes="192x192"
      type="image/png"
    />
    <link
      href="/favicon/favicon-32x32.png"
      rel="icon"
      sizes="32x32"
      type="image/png"
    />
    <link
      href="/favicon/favicon-96x96.png"
      rel="icon"
      sizes="96x96"
      type="image/png"
    />
    <link
      href="/manifest.json"
      rel="manifest"
    />
  </head>,
  <body>
    <main />
    NextScript
    <script
      dangerouslySetInnerHTML={
        {
          "__html": "
(function(a,b,c){var d=a.history,e=document,f=navigator||{},g=localStorage,h=encodeURIComponent,i=d.pushState,k=function(){return Math.random().toString(36)},l=function(){return g.cid||(g.cid=k()),g.cid},m=function(r){var s=[];for(var t in r)r.hasOwnProperty(t)&&void 0!==r[t]&&s.push(h(t)+"="+h(r[t]));return s.join("&")},n=function(r,s,t,u,v,ni,w,x){var z="https://www.google-analytics.com/collect",A=m({v:"1",ds:"web",aip:c.anonymizeIp?1:void 0,tid:b,cid:l(),t:r||"pageview",sd:c.colorDepth&&screen.colorDepth?
screen.colorDepth+"-bits":void 0,dr:e.referrer||void 0,dt:e.title,dl:e.location.origin+e.location.pathname+e.location.search,ul:c.language?(f.language||"").toLowerCase():void 0,de:c.characterSet?e.characterSet:void 0,sr:c.screenSize?(a.screen||{}).width+"x"+(a.screen||{}).height:void 0,vp:c.screenSize&&a.visualViewport?(a.visualViewport||{}).width+"x"+(a.visualViewport||{}).height:void 0,ec:s||void 0,ea:t||void 0,el:u||void 0,ev:v||void 0,exd:w||void 0,ni:ni?1:void 0,exf:"undefined"!=typeof x&&!1==
!!x?0:void 0});if(f.sendBeacon)f.sendBeacon(z,A);else{var y=new XMLHttpRequest;y.open("POST",z,!0),y.send(A)}};d.pushState=function(r){return"function"==typeof d.onpushstate&&d.onpushstate({state:r}),setTimeout(n,c.delay||10),i.apply(d,arguments)},n(),a.ma={trackEvent:function o(r,s,t,u,ni){return n("event",r,s,t,u,ni)},trackException:function q(r,s){return n("exception",null,null,null,null,null,r,s)}}})(window,"UA-67824860-3",{anonymizeIp:true,colorDepth:true,characterSet:true,screenSize:true,language:true});
",
        }
      }
    />
  </body>,
]
`;
